src/main/kotlin/com/chattriggers/ctjs/launch/plugin/networkManager.kt

```diff
@@ -0,0 +1,65 @@
+package com.chattriggers.ctjs.launch.plugin
+
+import me.falsehonesty.asmhelper.dsl.At
+import me.falsehonesty.asmhelper.dsl.InjectionPoint
+import me.falsehonesty.asmhelper.dsl.inject
+import me.falsehonesty.asmhelper.dsl.instructions.*
+
+fun injectNetworkManager() = inject {
+    className = "net/minecraft/network/NetworkManager"
+    methodName = "channelRead0"
+    methodDesc = "(Lio/netty/channel/ChannelHandlerContext;L$PACKET;)V"
+
+    methodMaps = mapOf("func_148833_a" to "processPacket")
+
+    at = At(InjectionPoint.INVOKE(
+            Descriptor(
+                    "net/minecraft/network/Packet",
+                    "processPacket",
+                    "(Lnet/minecraft/network/INetHandler;)V"
+            ),
+            0
+        )
+    )
+
+    insnList {
+        // create new cancellable event
+        createInstance(CANCELLABLE_EVENT, "()V")
+        val event = astore()
+
+        // get enum clientbound to filter out non-clientbound packets
+        getStatic("net/minecraft/network/EnumPacketDirection", "CLIENTBOUND", "Lnet/minecraft/network/EnumPacketDirection;")
+        val clientbound = astore()
+
+        // get "this" instance
+        aload(0)
+        // get the direction/sidedness of the packet
+        getField("net/minecraft/network/NetworkManager", "field_179294_g", "Lnet/minecraft/network/EnumPacketDirection;")
+        // push the clientbound enum onto stack
+        load(clientbound)
```

Comment:
No need to `astore` the enum here

```suggestion
        // get "this" instance
        aload(0)
        // get the direction/sidedness of the packet
        getField("net/minecraft/network/NetworkManager", "field_179294_g", "Lnet/minecraft/network/EnumPacketDirection;")
        // push the clientbound enum onto stack
        getStatic("net/minecraft/network/EnumPacketDirection", "CLIENTBOUND", "Lnet/minecraft/network/EnumPacketDirection;")
```